snap7/client.py

```diff
@@ -625,6 +625,13 @@ def as_read_area(self, area, dbnumber, start, size):
         :param start: offset to start writing
         :param size: number of units to read
         """
+        wordlen, type_, data = self._as_read_area_prepare(area, dbnumber, start, size)
+        result = self._library.Cli_AsReadArea(self._pointer, area, dbnumber, start, size, wordlen, byref(data))
+        check_error(result, context="client")
+        return bytearray(data)
+
+    @staticmethod
+    def _as_read_area_prepare(area, dbnumber, start, size):
```

Comment:
This is the beginning of the same code in both cases, just as private method and returns the stuff which is needed.
Same in the as_write_area case.